src/main/java/at/ac/tuwien/kr/alpha/common/BuiltinAtom.java

```diff
@@ -1,117 +1,164 @@
 package at.ac.tuwien.kr.alpha.common;
 
-import at.ac.tuwien.kr.alpha.grounder.NaiveGrounder;
+import at.ac.tuwien.kr.alpha.grounder.Substitution;
 import at.ac.tuwien.kr.alpha.grounder.parser.ParsedBuiltinAtom;
+import at.ac.tuwien.kr.alpha.grounder.parser.ParsedTerm;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Copyright (c) 2016, the Alpha Team.
  */
 public class BuiltinAtom implements Atom {
-	public final Term left;
-	public final Term right;
-	public final ParsedBuiltinAtom.BINOP binop;
+	private final List<Term> terms;
+	private final ParsedBuiltinAtom.BINOP binop;
+
+	protected BuiltinAtom(ParsedBuiltinAtom.BINOP binop, List<Term> terms) {
+		this.binop = binop;
+		this.terms = terms;
```

Comment:
Please add a check here to guarantee that a BuiltinAtom consists of only two terms. (Throw RuntimeException to abort.)